backend/app/pre_start.py

```
@@ -0,0 +1,49 @@
+)
+def init(db_engine) -> None:
+    """code to do the pre-start service"""
+    if db_engine is None:
+        logger.error("Database pool is not initialized")
+        raise RuntimeError("Database pool is not initialized")
+    try:
+        with db_engine.connection() as conn:
+            conn.execute("SELECT 1")
+        logger.info("Database connectivity check successful")
+    except Exception as e:
+        logger.error(f"Database connectivity check failed: {e}")
```

Ruff (0.15.1)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)